### PR TITLE
Privacy enhancement develop merge

### PR DESCRIPTION
**Pull request by UO283428**
DO NOT MERGE THIS BRANCH UNTIL GUILLERMO MERGES HIS.
Add place style changed, adapted to use the popups.
image
image

OverviewPage and pointInformation changed.
image

PrivacyComponent lets the user select the privacy by allowing to select public-private and to share it with friends.
image
image
Friends are not listed as the method for getting the username is not working properly.

MapInfo component shows the information of places and it is possible to change the privacy.
image
PointInformation page now is reusable to be used from different parts of the application and go back to the previous page.
image

### CODE REVIEW

**Review by uo282892 — APPROVED** [review]

File: webapp/src/components/social/UserPage.tsx [hunk 1]
```diff
@@ -54,16 +74,10 @@ export default class UserPage extends React.Component<UserPageProps, UserPageSta
...
-        this.maps = (<TableBody>
-            {this.testMaps.map((map) => (
-                <TableRow key={map.getName()} sx={{"&:last-child td, &:last-child th": {border: 0}}}>
-                    < TableCell component="th" scope="row">{map.getName()}</TableCell>
-                    <TableCell align="right">{map.getDescription()}</TableCell>
-                    <TableCell align="right">See map</TableCell>
-                </TableRow>
-            ))}
-        </TableBody>);
+    private async getMaps() {
```
Comment (Pelayo-Reguera) [inline webapp/src/components/social/UserPage.tsx:79]: This method is useless as it is

File: webapp/src/pages/MapInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+     * Returns the map information view
+     */
+    public render() {
+        if (this.state.goBack) {
+            return <UserStuff/>;
+        }
+        return (
+            <section className="my-stuff">
+                <div className="mapInformation">
+                    <h1>Title: {this.state.theMap.getName()}</h1>
```
Comment (Pelayo-Reguera) [inline webapp/src/pages/MapInfo.tsx:85]: Remove "Title:". Directly show the title

File: webapp/src/pages/MapInfo.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                    <h2>Description: {this.state.theMap.getDescription()}</h2>
```
Comment (Pelayo-Reguera) [inline webapp/src/pages/MapInfo.tsx:86]: Remove "Description:". Directly show the description

File: webapp/src/pages/MapInfo.tsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+                    <h2>Places:</h2>
+                    {/*For each place from the map an item will be a row with the name in a table, and a button to open a component PointInformation*/}
+                    {this.state.loadedPlaces &&
+                        <div className="places-buttons" >
+                            {this.state.theMap.getPlacemarks().map((placemark) => (
```
Comment (Pelayo-Reguera) [inline webapp/src/pages/MapInfo.tsx:91]: If it is possible, write a point before each name with the purpose of ahowing them like a list

File: webapp/src/pages/PointInformation.tsx [hunk 5]
```diff
@@ -73,70 +78,58 @@ export default class PointInformation extends React.Component<PointInformationPr
...
-            </section>
+            <Modal open={this.state.open} onClose={() => {
+                this.setState(({open: false}));
+                this.props.onBack(this.props.prevComponent);
+            }}>
+                <ModalDialog className="custom-modal-dialog">
+                    <ModalClose/>
+                    <section className="pointInfo" /*style={{overflow: "scroll"}}*/>
+                        <div className="pointInformation">
+                            <h1>Title: {this.point.title}</h1>
```
Comment (Pelayo-Reguera) [inline webapp/src/pages/PointInformation.tsx:106]: Remove "Title:". Directly show the title

**Review by Pelayo-Reguera — DISMISSED** [review]

**Review by UO283069 — APPROVED** [review]

**Review by Pelayo-Reguera — APPROVED** [review]

**Pull request merged**